lib/remote/actionshandler.cpp

```diff
@@ -100,12 +100,31 @@ bool ActionsHandler::HandleRequest(
 	}
 
 	int statusCode = 500;
+	std::set<int> statusCodes, nonOkStatusCodes;
 
 	for (const Dictionary::Ptr& res : results) {
-		if (res->Contains("code") && res->Get("code") >= 200 && res->Get("code") <= 299) {
-			statusCode = 200;
-			break;
+		if (!res->Contains("code")) {
+			continue;
 		}
+
+		auto getCode = res->Get("code");
+
+		if (getCode >= 200 && getCode <= 299) {
+			statusCodes.insert(getCode);
```

Comment:
Since this does not contain all status codes, I'd call the set `okStatusCodes` instead.